packages/engine/src/interaction/functions/interactUI.ts

```diff
@@ -0,0 +1,237 @@
+import { Entity } from '../../ecs/classes/Entity'
+import { MathUtils, Quaternion, Vector3, Object3D } from 'three'
+import { addComponent, getComponent, hasComponent, defineQuery } from '../../ecs/functions/ComponentFunctions'
+import { Object3DComponent } from '../../scene/components/Object3DComponent'
+import { TransformComponent } from '../../transform/components/TransformComponent'
+import { XRUIComponent } from '@xrengine/engine/src/xrui/components/XRUIComponent'
+import { InteractableComponent } from '../components/InteractableComponent'
+import { RenderedComponent } from '../../scene/components/RenderedComponent'
+import { LocalInputTagComponent } from '../../input/components/LocalInputTagComponent'
+import { AvatarComponent } from '../../avatar/components/AvatarComponent'
+import { FollowCameraComponent } from '../../camera/components/FollowCameraComponent'
+
+import { Engine } from '../../ecs/classes/Engine'
+import { LoadGLTF } from '@xrengine/engine/src/assets/functions/LoadGLTF'
+
+import { hideInteractText, showInteractText, createInteractText } from './interactText'
+import { createInteractiveModalView, connectCallback } from '../ui/InteractiveModalView'
+
+import Hls from 'hls.js'
+import isHLS from '@xrengine/engine/src/scene/functions/isHLS'
+
+/**
+ * @author Ron Oyama <github.com/rondoor124>
+ */
+
+const upVec = new Vector3(0, 1, 0)
+const localUserQuery = defineQuery([LocalInputTagComponent, AvatarComponent])
+const xrUIQuery = defineQuery([XRUIComponent, Object3DComponent])
+
+export const InteactiveUI = new Map<Entity, ReturnType<typeof createInteractiveModalView>>()
+
+//TODO: Create interactive UI
+export const createInteractUI = (entity: Entity) => {
+  const interactiveComponent = getComponent(entity, InteractableComponent)
+  if (getInteractUI(entity) || !interactiveComponent || !interactiveComponent.data) return
+
+  //create interactive view
+  interactiveComponent.data.interactionUserData = {}
+  interactiveComponent.data.interactionUserData.entity = entity
+  const ui = createInteractiveModalView(interactiveComponent.data)
+  InteactiveUI.set(entity, ui)
+
+  //set transform
+  const transform = getComponent(entity, TransformComponent)
+  addComponent(ui.entity, TransformComponent, {
+    position: new Vector3(transform.position.x, transform.position.y + 2.5, transform.position.z),
+    rotation: new Quaternion(),
+    scale: new Vector3(1, 1, 1)
+  })
+
+  //create text
+  const interactTextEntity = createInteractText(interactiveComponent.data.interactionText)
+
+  const timer = setInterval(() => {
```

Comment:
Using setInterval seems like an anti-pattern; It would be better to use an enter() query w/ XRUIComponent in the InteractionSystem